TIC-114 Front-End Active Events Component Rework

jonathantnguyen opened the pull request:
### Description
Switching to drop down selection to Group Toggle MUI design component that is it's own file for re-usability.

Before:
![image](https://github.com/WonderTix/WonderTix/assets/15915111/dff78cd9-9768-468f-b8da-ce8132ac69b9)

After:
![image](https://github.com/WonderTix/WonderTix/assets/15915111/f6c5aaa5-3fd2-43e4-afa5-1c6a4e46c2de)
![image](https://github.com/WonderTix/WonderTix/assets/15915111/4d08a658-b7e2-4352-b437-6921efcb2222)


### Risks
None currently, just front-end work.

### Validation
Manual Testing

### Issue
[*Link to corresponding issue*](https://pdx-hkaus.atlassian.net/browse/TIC-114)

### Operating System
Win10/Chrome

henrykaus commented on client/src/components/Ticketing/GroupeToggle.tsx:
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
```
Usually we just want to import as little as possible, so we probably want to do the specific things we're importing, even though I know thats what the mui docs look like

jonathantnguyen replied:
On line 6, I use `const [filterAs, setFilter] = React.useState('Active');`. Is this still unneeded? I result in code being underlined in red.

jonathantnguyen replied:
On line 6, I use ` const [filterAs, setFilter] = React.useState('Active');`. Is it still unneeded? Currently, removing the import results in code being underlined in red.

henrykaus replied:
Well, you wouldn't want to remove the import but something more along the lines of `import {useState} from 'react';`

henrykaus replied:
And that import statement would also need to include anything else that is imported from `'react'` like MouseEvent

henrykaus commented on client/src/components/Ticketing/GroupeToggle.tsx:
```diff
@@ -0,0 +1,45 @@
+      value={filterAs}
+      exclusive
+      onChange={handleFilterChange}
+      aria-label='filterSetting'
```
`aria-label`s are essentially supposed to be read outloud by a screen reader so it should be something understandable yet simple. I'd probably recommend something like 'activeness toggle filter'

henrykaus commented on client/src/components/Ticketing/GroupeToggle.tsx:
```diff
@@ -0,0 +1,45 @@
+import ToggleButton from '@mui/material/ToggleButton';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+
+const GroupToggle = () => {
```
Since this is specifically for toggling between activeness for ticketing things, I'd be a little more specific with the component name rather than a generic GroupToggle, and then make the filename match

jonathantnguyen replied:
fixed

henrykaus commented on client/src/components/Ticketing/GroupeToggle.tsx:
```diff
@@ -0,0 +1,45 @@
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+
+const GroupToggle = () => {
+  const [filterAs, setFilter] = React.useState('Active');
```
For consistency sake, the second item in the use state should be named the same except with a prepended `set`. So this can either be `[filterAs, setFilterAs]` or `[filter, setFilter]`

henrykaus replied:
I also imagine that this should exactly match the value fields below, so `'active'`. This way `active` will be selected by default when the page loads

henrykaus replied:
This field would also be a good prop to add to this component. Something like `defaultValue` that could only be `active`, `inactive` or `all` and then gets thrown into the `useState` as the baseline

jonathantnguyen replied:
added/fixed

0Tendo commented on client/src/components/Ticketing/ticketingmanager/Season/SeasonInstancesPage.tsx:
```diff
@@ -2,6 +2,7 @@ import React, {useEffect, useState} from 'react';
 import {useNavigate} from 'react-router-dom';
 import {titleCase} from '../../../../utils/arrays';
 import {getSeasonImage, formatSeasonDate, SeasonImage} from './seasonUtils';
+import GroupToggle from '../../GroupeToggle';
```
Just curious, any reason for some being Group and some being Groupe?

jonathantnguyen replied:
Thanks for catching the typo. Fixed.

0Tendo reviewed (approved):
I like this change. Good idea making it a separate component. We should incorporate this into the doorlist once we get the backend working.

henrykaus reviewed (commented):
Just a couple comments, it overall looks good and functions great. My only stylistic suggestion would be to make the text a little bolder, it looks a little thin at the moment and makes the gray text hard to read. I think grey is good, but the text should be a little bolder if possible

jonathantnguyen commented:
New changes include:

- FontWeight: Bold
- Added defaultValue to component & associated files which need to include "defaultValue"
- Renamed files, component name, and labels.

henrykaus commented on client/src/components/Ticketing/GroupToggle.tsx:
```diff
@@ -44,5 +45,9 @@ const ShowingActivenessToggle =(defaultValue) => {
   );
 };
 
+ShowingActivenessToggle.propTypes = {
+  defaultValue: PropTypes.string.isRequired,
+};
+
```
I love this library so I'm glad you're using it

henrykaus reviewed (approved):
Thanks for implementing all of the changes! This component looks good
